edb/server/protocol/auth_ext/__init__.py

```diff
@@ -0,0 +1,86 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2022-present MagicStack Inc. and the EdgeDB authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import http
+import json
+import urllib.parse
+
+from edb import errors
+from edb.common import debug
+from edb.common import markup
+
+from . import oauth
+
+
+async def handle_request(
+    request, response, db, extension_base_path: str, args: list[str]
+):
+    try:
+        # Set up routing to the appropriate handler
+        if args[0] == "authorize":
+            provider_name: str | None = _get_search_param(
+                request.url.query.decode("ascii"), "provider"
+            )
+            if provider_name is None:
+                raise errors.BackendError(
+                    "No provider specified in URL search parameters."
+                )
+
+            redirect_uri = f"{extension_base_path}/callback"
+
+            return oauth.redirect_to_auth_provider(
+                response=response,
+                redirect_uri=redirect_uri,
+                iss=request.host.decode(),
```

Comment:
As this might be related to security, I don't think it's the best idea to blindly use the `Host` header as token issuer. I don't know but we probably want to have a separate optional config for it, and falls back to `Host/X-Forwarded-*`.

Reply:
Oh actually, this should be the server's netloc (e.g. `https://localhost:8080`). We should pass this into the handler I think, or maybe derive it from the `base_extension_path`.

Reply:
Also, I'm glad you brought this up because I need to remember to check the `iss` in the callback state check, too. 🙏

Reply:
Turns out, the issuer is really the full `base_extension_path` since a single server might be issuing tokens for multiple databases and we don't want to consider them all a single issuer.

⚡ [`55cceb8` (#5816)](https://github.com/edgedb/edgedb/pull/5816/commits/55cceb8b712498e5c4c439e843b3142afef266e9)

How does that look?

Reply:
ahh good point, makes sense to me!